ui/tests/e2e/fixtures/environmentsPage.js

```diff
@@ -0,0 +1,221 @@
+import { expect } from '@playwright/test';
+import { ENV } from '../env';
+
+export class EnvironmentPage {
+  /**
+   * @param {import('@playwright/test').Page} page
+   */
+  constructor(page) {
+    this.page = page;
+  }
+  async navigate() {
+    await this.page.goto(`${ENV.MESHERY_SERVER_URL}/management/environments`);
+  }
+  async fillInput(locator, value) {
+    await locator.fill(value);
+  }
+
+  async goToProfiles() {
+    await this.page.goto(`${ENV.MESHERY_SERVER_URL}/management/environments`);
+    await expect(this.page).toHaveURL(`${ENV.MESHERY_SERVER_URL}/management/environments`);
+  }
+
+  async goToConnections() {
+    await this.page.goto(`${ENV.MESHERY_SERVER_URL}/management/connections`);
+    await expect(this.page).toHaveURL(`${ENV.MESHERY_SERVER_URL}/management/connections`);
+  }
+
+  async doesProfileExist(environmentName) {
+    await this.goToProfiles();
+    return this.page
+      .locator(`text=${environmentName}`)
+      .count()
+      .then((count) => count === 1);
+  }
+  async createEnvironmentProfile(environmentName) {
+    if (!(await this.doesProfileExist(environmentName))) {
+      await this.page.getByRole('button', { name: 'Create' }).click();
+      await this.fillInput(this.page.locator('#Environment_name'), `${environmentName}`);
+      await this.fillInput(
+        this.page.locator('#Environment_description'),
+        'Created profile for playwright tests',
+      );
+      await this.page.getByRole('button', { name: 'Save', exact: true }).click();
+      // Check for notification visibility
+      const notification = await this.page.locator(`text=Environment "${environmentName}" created`);
+      await expect(notification).toBeVisible();
+      await this.page.waitForTimeout(8000);
+    } else {
+      console.log(`Profile "${environmentName}" already exists.`);
+    }
+  }
+
+  async assignConnections(environmentName) {
+    await expect(this.page).toHaveURL(`${ENV.MESHERY_SERVER_URL}/management/environments`);
+    await this.page.waitForSelector(`text=${environmentName}`, { state: 'visible' });
+    await this.page.evaluate(() => {
+      const button = [...document.querySelectorAll('button')].find((b) =>
+        b.textContent.includes('Assigned Connections'),
+      );
+      if (button) {
+        button.setAttribute('data-testid', 'assigned-connections-button');
```

Comment:
Can we not do dependencies injection like this? Playwright have filtering mechanism maybe this can help

https://playwright.dev/docs/locators#filtering-locators

Reply:
Sure I'll fix that